pkg/assembler/clients/operations/path.graphql

```diff
@@ -0,0 +1,150 @@
+#
+# Copyright 2023 The GUAC Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# NOTE: This is experimental and might change in the future!
+
+# Exposes GraphQL queries to retrieve GUAC graph connectivity data
+
+query Path($subject: ID!, $target: ID!, $maxPathLength: Int!) {
+  path(subject: $subject, target: $target, maxPathLength: $maxPathLength) {
+    __typename
+    ... on Package {
+      ... allPkgTree
+    }
+    ... on Source {
+      ... allSourceTree
+    }
+    ... on Artifact {
+      ...allArtifactTree
+    }
+    ... on Builder {
+      ...allBuilderTree
+    }
+    ... on OSV {
+      ...allOSVTree
+    }
+    ... on CVE {
+      ...allCveTree
+    }
+    ... on GHSA {
+      ...allGHSATree
+    }
+    ... on CertifyScorecard {
+      ...allCertifyScorecard
+    }
+    ... on CertifyPkg {
+      ...allCertifyPkg
+    }
+    ... on IsOccurrence {
+      ...allIsOccurrencesTree
+    }
+    ... on IsDependency {
+      ...allIsDependencyTree
+    }
+    ... on HasSLSA {
+      ...allSLSATree
+    }
+    ... on CertifyBad {
+      ...allCertifyBad
+    }
+    ... on HashEqual {
+      ...allHashEqualTree
+    }
+    ... on HasSBOM {
+      ...allHasSBOMTree
+    }
+    ... on HasSourceAt {
+      ...allHasSourceAt
+    }
+    ... on CertifyVuln {
+      ...allCertifyVuln
+    }
+    ... on IsVulnerability {
+      ...allIsVulnerability
+    }
+    ... on CertifyVEXStatement {
+      ...allCertifyVEXStatement
+    }
+    ... on Builder {
+      ...allBuilderTree
+    }
+  }
+}
+
+query Neighbors($node: ID!) {
+  neighbors(node: $node) {
+    __typename
```

Comment:
qq: what is the typename here again?

Reply:
that's how graphQL knows how to parse: 	Typename *string `json:"__typename"`

Reply:
It's the node name, the name from GraphQL (so, Package, Artifact, etc.) This is how you know what type was actually used in the union